packages/caliper-core/lib/worker/workload/declarative/value-providers/variable-reference-value-provider.js

```diff
@@ -0,0 +1,55 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+const ValueProviderInterface = require('./value-provider-interface');
+
+/**
+ * Class with implementation of VariableReferenceValueProvider
+ */
+class VariableReferenceValueProvider extends ValueProviderInterface {
+    /**
+     * Initialize an instance of VariableReferenceValueProvider
+     * @param {object} options Options for configured parameter generation
+     * @param {object} variables Caliper provided variables
+     * @param {object} parameters User provided variables
+     * @param {object} valueProviderFactory ValueProviderFactory object reference
+     */
+    constructor(options, variables, parameters, valueProviderFactory) {
+        super(options, variables, parameters, valueProviderFactory);
+
```

Comment:
It should be checked whether `this.options` is undefined, and throw an error if it is.